### DISCUSSION TITLE
ESP32C6 and A2DP-Source

### DISCUSSION THREAD

**Discussion by mjmeans** [pschatzmann/arduino-audio-tools#1759] [Q&A] [answered]
So, I need to be able to send one of several short audio clips (like 2 seconds long) to a BT speaker or earbug. I selected the XIAO-ESP32C6 because it has 4 MB of memory and that is large enough for the program and sound clips, and without needing an SD card or the space required for an SD card interface on the PCB that I would never use.

I saw the article on https://www.pschatzmann.ch/home/2021/11/02/streaming-mp3-files-to-a-bluetooth-speaker/ and tried it with my board. But the https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication/a2dp/player-sdfat-a2dp sample doesn't compile and reports the error "    32 | #  error "ESP32C3, ESP32S2, ESP32S3 do not support A2DP"".

Why does the error message display for this ESP32C6 board, but not state ESP32C6 as a problem? Is this a bug?

I'm not looking to do A2DP-Sink, but A2DP-Source. Does ESP32C6 actually not have the ability to do A2DP-Source?

Why would an ESP32 board with an I2S interface be needed if I never want to play audio to wired speakers?

Assuming that SP32C6 actually can't do what I want at all, what is the smallest ESP32 board that can do what I want (play audio clips to a BT speaker or earbud), and has a built-in battery charger?

**Comment by pschatzmann** [comment, marked as answer]
Please read the [Wiki of the A2DP projecdt](https://github.com/pschatzmann/ESP32-A2DP/wiki/Support-for-esp32-c3%3F) or better the Data Sheet or your processor!
